### PR TITLE
Feat/dev 159 detect secrets

### PR DESCRIPTION
**Pull request by asurinsaka**
DEV-159 add pre-commit and detect-secrets

We would like to use pre-commit to perform various checks for us. 
Detecting secrets before committing is one use-case.

### CODE REVIEW

**Review by ProfOak — APPROVED** [review]

**Review by philtom-ctds — CHANGES_REQUESTED** [review]
Need to fix and update master before merging this pr.  I've opened prs to address master.  Blocking this pr for now

**Comment by asurinsaka** [comment]
> Need to fix and update master before merging this pr. I've opened prs to address master. Blocking this pr for now

I do not need to do anything for this one, do I?

**Review by philtom-ctds — APPROVED** [review]

**Pull request merged**
